src/mlpack/methods/ann/layer/lookup_impl.hpp

```diff
@@ -21,30 +21,39 @@ namespace ann /** Artificial Neural Network. */ {
 
 template <typename InputDataType, typename OutputDataType>
 Lookup<InputDataType, OutputDataType>::Lookup(
-    const size_t inSize,
-    const size_t outSize) :
-    inSize(inSize),
-    outSize(outSize)
+    const size_t vocabSize,
+    const size_t embeddingSize) :
+    vocabSize(vocabSize),
+    embeddingSize(embeddingSize)
 {
-  weights.set_size(outSize, inSize);
+  weights.set_size(embeddingSize, vocabSize);
 }
 
 template<typename InputDataType, typename OutputDataType>
 template<typename eT>
 void Lookup<InputDataType, OutputDataType>::Forward(
     const arma::Mat<eT>& input, arma::Mat<eT>& output)
 {
-  output = weights.cols(arma::conv_to<arma::uvec>::from(input) - 1);
+  const size_t seqLength = input.n_rows;
+  const size_t batchSize = input.n_cols;
+
+  output.set_size(embeddingSize * seqLength, batchSize);
+
+  for (size_t i = 0; i < batchSize; ++i)
+  {
+    output.col(i) = arma::vectorise(weights.cols(
+        arma::conv_to<arma::uvec>::from(input.col(i)) - 1));
+  }
 }
 
 template<typename InputDataType, typename OutputDataType>
 template<typename eT>
 void Lookup<InputDataType, OutputDataType>::Backward(
     const arma::Mat<eT>& /* input */,
-    const arma::Mat<eT>& gy,
-    arma::Mat<eT>& g)
+    const arma::Mat<eT>& /* gy */,
+    arma::Mat<eT>& /* g */)
 {
-  g = gy;
+  // Nothing to do here.
```

Comment:
Correct me if I'm wrong. This function isn't meant to be used since this is the first layer of the network. Perhaps in this case it could be helpful to insert `Log::Fatal()` here or something.

Reply:
Yes. `Backward()` can be called but nothing needs to be done here. Also, the `FFN` by its property will call `Backward()` function for all layers and if we put `Log::Fatal()`, it will break. But we don't want that, we just want to skip it.

Reply:
I thought it skips the first layer:
```c++
  for (size_t i = 2; i < network.size(); ++i)
  {
    boost::apply_visitor(BackwardVisitor(boost::apply_visitor(
        outputParameterVisitor, network[network.size() - i]),
        boost::apply_visitor(deltaVisitor, network[network.size() - i + 1]),
        boost::apply_visitor(deltaVisitor, network[network.size() - i])),
        network[network.size() - i]);
  }
```
I'll check it. Perhaps it calls `Backward()` somewhere else.

Reply:
Ohh...my mistake. I was putting `i = network.size()` for last iteration and then assumed it will visit up to `network[0]`. I think then it's okay to put `Log::Fatal()` or at least `Log::Warn()` when someone tries to use it as an intermediate layer. Sorry for the confusion!

Also, I got a question now. If `Backward()` doesn't visit the first layer, how can we create a network with a single layer?
```cpp
FFN<MeanSquaredError<>, RandomInitialization> model;
model.Add<Lookup<>>(vocabSize, embeddingSize);
```

Reply:
Ahh... got it. Probably we need to use `IdentityLayer` at the end.

```cpp
FFN<MeanSquaredError<>, RandomInitialization> model;
model.Add<Lookup<>>(vocabSize, embeddingSize);
model.Add<Identity layer<>>();
```

Reply:
> If Backward() doesn't visit the first layer, how can we create a network with a single layer?

I think it's not supposed to create such networks. They usually have a classifier at the end.

Reply:
Correct!